ts/features/idpay/initiative/configuration/components/InstrumentEnrollmentSwitch.tsx

```diff
@@ -0,0 +1,167 @@
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import { Badge, ListItem, View } from "native-base";
+import { default as React } from "react";
+import { StyleSheet } from "react-native";
+import {
+  InstrumentDTO,
+  StatusEnum
+} from "../../../../../../definitions/idpay/wallet/InstrumentDTO";
+import { Body } from "../../../../../components/core/typography/Body";
+import { H4 } from "../../../../../components/core/typography/H4";
+import { LabelSmall } from "../../../../../components/core/typography/LabelSmall";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import Switch from "../../../../../components/ui/Switch";
+import I18n from "../../../../../i18n";
+import { Wallet } from "../../../../../types/pagopa";
+import { useIOBottomSheetModal } from "../../../../../utils/hooks/bottomSheet";
+import { instrumentStatusLabels } from "../../../common/labels";
+
+type InstrumentEnrollmentSwitchProps = {
+  wallet: Wallet;
+  status?: InstrumentDTO["status"];
+  isDisabled?: boolean;
+  onEnrollInstrument?: (walletId: number) => void;
+  onDeleteInstrument?: (walletId: number) => void;
+};
+
+type InstrumentInfo = {
+  logo: JSX.Element;
+  maskedPan: string;
+};
+
+/**
+ * A component to enable/disable the enrollment of an instrument
+ */
+const InstrumentEnrollmentSwitch = (props: InstrumentEnrollmentSwitchProps) => {
+  const { wallet, status, isDisabled, onEnrollInstrument, onDeleteInstrument } =
+    props;
+
+  const [switchStatus, setSwitchStatus] = React.useState(
+    status === StatusEnum.ACTIVE
+  );
+
+  const enrollmentBottomSheetModal = useIOBottomSheetModal(
```

Comment:
This was a tricky part and I was not sure about the ideal solution tbh. 
The problem is that if the user abort the instrument enrollment and closes the modal we need to revert the switch to its previous state. 
But I agree that it would be better to find ways to use a single bottomsheet for all switches. 
I'll work on it! 
Thanks